_posts/2019-05-24-varying-slopes/varying_slopes.Rmd

```diff
@@ -0,0 +1,748 @@
+Or using the technical term, in the latter case we hope for the model to _shrink_ its estimates toward the overall mean more noticeably than in the former.
+
+This type of information sharing is already very useful, but it gets better. The tadpole model is a _varying intercepts_ model, as McElreath calls it (or _random intercepts_, as it is sometimes -- confusingly -- called ^[cf. [the Wikipedia article on multilevel models](https://en.wikipedia.org/wiki/Multilevel_model) for a collection of terms encountered when dealing with this subject, and e.g. [Gelman's](https://statmodeling.stat.columbia.edu/2005/01/25/why_i_dont_use/) dissection of various ways _random effects_ are defined]) -- _intercepts_ referring to the way we make predictions for entities (here: tanks), with no predictor variables present. So if we can pool information about intercepts, why not pool information about _slopes_ as well? This will allow us to, in addition, make use of _relationships_ between variables learnt on different entities in the training set.
+
+So as you might have guessed by now, _varying slopes_ (or _random slopes_, if you will) is the topic of today's post. Again, we take up an example from McElreath's book, and show how to accomplish the same thing with `tfprobability`.
+
+## Coffee, please
+
+Unlike the tadpole case, this time we work with simulated data. This is the data McElreath uses to introduce the _varying slopes_ modeling technique; he then goes on and applies it to one of the book's most featured datasets, the _pro-social_ (or indifferent, rather!) chimpanzees. For today, we stay with the simulated data for two reasons: First, the subject matter per se is non-trivial enough; and second, we want to keep careful track of what our model does, and whether its output is sufficiently close to the results McElreath obtained from _Stan_ ^[We won't overload this post by explicitly comparing results here, but we did that when writing the code.]. 
+
+So, the scenario is this. ^[Disclaimer: We have not verified whether this is an adequate model of the world, but it really doesn't matter either.] Cafés vary in how popular they are. In a popular café, when you order coffee, you're likely to _wait_. In a less popular café, you'll likely be served much faster. That's one thing.
+Second, all cafés tend to be more crowded in the mornings than in the afternoons. Thus in the morning, you'll wait longer than in the afternoon -- this goes for the popular as well as the less popular cafés.
+
+In terms of intercepts and slopes, we can picture the morning waits as intercepts, and the resultant afternoon waits as arising due to the slopes of the lines joining each morning and afternoon wait, respectively.
```

Comment:
Is there a visual here? I ask mainly because "lines" could also refer to lines in which people are waiting; so, it might be helpful to see a generalized example chart.